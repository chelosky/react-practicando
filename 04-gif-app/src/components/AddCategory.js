import React, { useState } from 'react'
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories  }) => {

    const [value, setValue] = useState('');

    const handleInputChange = (e) => {
        setValue(e.target.value);
    }

    const handleSubmit = (e) => {
        e.preventDefault();

        if(value.trim().length > 2){
            setCategories( item => [ value, ...item ] );
            setValue('');
        }
    }

    return (
        <form onSubmit={ handleSubmit }>
            <input
                type="text"
                value={ value }
                onChange={ handleInputChange }
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}