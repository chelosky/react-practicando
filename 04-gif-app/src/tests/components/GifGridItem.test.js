import React from "react"
import { shallow } from "enzyme";
import {GifGridItem} from '../../components/GifGridItem';

describe('GIFGRIDITEM', () => {

    const title = 'Titulo xd';
    const url = 'https://localhost/algo.jpg';
    const wrapper = shallow( <GifGridItem title={title} url={url} /> );

    test('should be created', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
    test('should have a p tag with the title', () => {
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })
    
    test('should img have url in src and the title like alt title', () => {
        const img = wrapper.find('img');
        // console.log(img.props().src);
        // console.log(img.prop('src'));
        // console.log(img.props().alt);
        expect(img.prop('src').trim()).toBe(url);
        expect(img.prop('alt').trim()).toBe(title);
    })
    
    test('should div have a animate__fadeIn class', () => {
        const div = wrapper.find('div');
        const className = div.prop('className');
        expect(className.includes('animate__fadeIn')).toBe(true);
    })
    

})