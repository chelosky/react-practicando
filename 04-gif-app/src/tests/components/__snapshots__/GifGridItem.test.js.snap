// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIFGRIDITEM should be created 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo xd"
    src="https://localhost/algo.jpg"
  />
  <p>
     
    Titulo xd
     
  </p>
</div>
`;
